plugins/CMakeLists.txt

```diff
@@ -1,4 +1,102 @@
+cmake_minimum_required(VERSION 3.16...3.25)
+
 option(ENABLE_PLUGINS "Enable building OBS plugins" ON)
+
+if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+
+  if(NOT ENABLE_PLUGINS)
+    set_property(GLOBAL APPEND PROPERTY OBS_FEATURES_DISABLED "Plugin Support")
+  endif()
+
+  set_property(GLOBAL APPEND PROPERTY OBS_FEATURES_ENABLED "Plugin Support")
+
+  macro(check_obs_browser)
+    if(OS_MACOS
+       OR (OS_WINDOWS AND CMAKE_GENERATOR_PLATFORM MATCHES "(Win32|x64)")
+       OR (OS_LINUX AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "[aA]?[aA][rR]([mM]|[cC][hH])64"))
```

Comment:
As currently written, this would also match things like:
* aarm64
* arch64

Might be cleaner as something like this, unless you want to avoid intermediate variables at all costs:
```
string(TOLOWER CMAKE_SYSTEM_PROCESSOR _sys_cpu_arch)
[...]
OR (OS_LINUX AND NOT _sys_cpu_arch MATCHES "(aarch64|arm64)"))
```

Reply:
That would match, but I'm also not aware of `uname -m` ever yielding any value but `aarch64` or `arm64`.

_However_ given that Windows arm64 is the only platform without a pre-built CEF by now, it's probably easier to limit the check to that.

Reply:
Sure, I'm aware of the values we expect for `uname -m`, but as-written, the regex is needlessly complicated and confusing. I prefer that it be immediately clear what we're looking for.